### PR TITLE
feat(all): Add example chain implementation

### PR DESCRIPTION
**Pull request by MalteHerrmann**
This PR adds a reduced chain implementation, that extends Cosmos SDK's [simapp](https://github.com/cosmos/cosmos-sdk/blob/v0.47.13/simapp/README.md) in using the evmOS modules: `erc20`, `evm` and `feemarket`.
It comes with its own equivalent of the `local_node.sh` script which is used in the Evmos repository for simple testing.

## Summary by CodeRabbit

- **New Features**
	- Introduced a comprehensive setup for a local blockchain testnet, allowing developers to easily create and manage multi-validator environments.
	- Added a `README.md` for documentation on the example chain, detailing configurations and usage instructions.
	- Integrated a command-line interface for managing blockchain operations through a new CLI application.
	- Introduced functions for exporting application state and validator management essential for the genesis block.

- **Bug Fixes**
	- Minor formatting adjustments made to several files for improved readability.

- **Documentation**
	- Added extensive documentation for the example chain and its configurations, enhancing user understanding.

- **Chores**
	- Created a new Bash script to automate local node setup and configuration tasks.
- **Improvements**
	- Updated the changelog to include references to recent enhancements and example implementations.

### CODE REVIEW

File: example_chain/app.go [hunk 1]
```diff
@@ -0,0 +1,702 @@
...
+}
+
+// GetMaccPerms returns a copy of the module account permissions
+//
+// NOTE: This is solely to be used for testing purposes.
+func GetMaccPerms() map[string][]string {
+	dupMaccPerms := make(map[string][]string)
+	for k, v := range maccPerms {
+		dupMaccPerms[k] = v
+	}
```
[the remark this answers is not in the document]
Comment (MalteHerrmann) [reply example_chain/app.go:778]: keys are being sorted afterwards

File: example_chain/local_node.sh [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+	sed -i.bak 's/"max_deposit_period": "172800s"/"max_deposit_period": "30s"/g' "$GENESIS"
+	sed -i.bak 's/"voting_period": "172800s"/"voting_period": "30s"/g' "$GENESIS"
+
+	# set custom pruning settings
+	sed -i.bak 's/pruning = "default"/pruning = "custom"/g' "$APP_TOML"
+	sed -i.bak 's/pruning-keep-recent = "0"/pruning-keep-recent = "2"/g' "$APP_TOML"
+	sed -i.bak 's/pruning-interval = "0"/pruning-interval = "10"/g' "$APP_TOML"
+
+	# Allocate genesis accounts (cosmos formatted addresses)
+	osd genesis add-genesis-account "$(osd keys show "$VAL_KEY" -a --keyring-backend "$KEYRING" --home "$HOMEDIR")" 100000000000000000000000000aevmos --keyring-backend "$KEYRING" --home "$HOMEDIR"
```
Comment (GAtom22) [inline example_chain/local_node.sh:172]: Nit: change the denom
```suggestion
	osd genesis add-genesis-account "$(osd keys show "$VAL_KEY" -a --keyring-backend "$KEYRING" --home "$HOMEDIR")" 100000000000000000000000000aos --keyring-backend "$KEYRING" --home "$HOMEDIR"
```

File: example_chain/local_node.sh [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+	echo "$USER1_MNEMONIC" | osd keys add "$USER1_KEY" --recover --keyring-backend "$KEYRING" --algo "$KEYALGO" --home "$HOMEDIR"
+	echo "$USER2_MNEMONIC" | osd keys add "$USER2_KEY" --recover --keyring-backend "$KEYRING" --algo "$KEYALGO" --home "$HOMEDIR"
+	echo "$USER3_MNEMONIC" | osd keys add "$USER3_KEY" --recover --keyring-backend "$KEYRING" --algo "$KEYALGO" --home "$HOMEDIR"
+	echo "$USER4_MNEMONIC" | osd keys add "$USER4_KEY" --recover --keyring-backend "$KEYRING" --algo "$KEYALGO" --home "$HOMEDIR"
+
+	# Set moniker and chain-id for the example chain (Moniker can be anything, chain-id must be an integer)
+	osd init $MONIKER -o --chain-id "$CHAINID" --home "$HOMEDIR"
+
+	# Change parameter token denominations to desired value
+	jq '.app_state["staking"]["params"]["bond_denom"]="aevmos"' "$GENESIS" >"$TMP_GENESIS" && mv "$TMP_GENESIS" "$GENESIS"
```
Comment (GAtom22) [inline example_chain/local_node.sh:118]: nit
```suggestion
	jq '.app_state["staking"]["params"]["bond_denom"]="aos"' "$GENESIS" >"$TMP_GENESIS" && mv "$TMP_GENESIS" "$GENESIS"
```

File: example_chain/local_node.sh [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+	if [[ $1 == "pending" ]]; then
+		echo "pending mode is on, please wait for the first block committed."
+	fi
+fi
+
+# Start the node
+osd start \
+	--log_level $LOGLEVEL \
+	--minimum-gas-prices=0.0001aevmos \
```
Comment (GAtom22) [inline example_chain/local_node.sh:201]: nit
```suggestion
	--minimum-gas-prices=0.0001aos \
```

**Review by GAtom22 — APPROVED** [review]
LGTM! Great work @MalteHerrmann!!

Left some nit comments.

File: example_chain/local_node.sh (hunk 2 above)
Comment (MalteHerrmann) [reply example_chain/local_node.sh:172]: I wanted to use `aos` but that makes `os` the display denomination and only >3 characters are supported so I stuck with `aevmos` - theoretically that's going to be `aevmOS` haha

File: example_chain/ante/evm_handler.go [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+	evmKeeper          anteinterfaces.EVMKeeper
+	distributionKeeper anteinterfaces.DistributionKeeper
+	stakingKeeper      anteinterfaces.StakingKeeper
+	maxGasWanted       uint64
+}
+
+// NewEVMMonoDecorator creates the 'mono' decorator, that is used to run the ante handle logic
+// for EVM transactions on the chain.
+//
+// TODO: make partners not have to duplicate this in their repo
```
[the remark this answers is not in the document]
Comment (MalteHerrmann) [reply example_chain/ante/evm_handler.go:52]: yes please create the corresponding issue

**Pull request merged**